code/game/objects/items/weapons/twohanded.dm

```diff
@@ -616,12 +622,12 @@
 		return ..()
 
 /obj/item/twohanded/chainsaw/wield() //you can't disarm an active chainsaw, you crazy person.
-	..()
-	flags |= NODROP
+	if((. = ..()))
```

Comment:
This is just going to check if the `. = ..()` operation complete successfully (which it always will). Make this a 
```
. = ..()
if(.)
```
check